src/app/index.ts

```diff
@@ -160,37 +196,40 @@ module.exports = yo.extend({
         default: 'Excel',
         choices: jsonData.getHostTemplateNames().map(host => ({ name: host, value: host })),
         when: (this.options.host == null || this.options.host != null && !jsonData.isValidInput(this.options.host, true /* isHostParam */))
-        && !isExcelFunctionsProject
+          && !isExcelFunctionsProject
       }];
       let answerForHost = await this.prompt(askForHost);
       let endForHost = (new Date()).getTime();
       let durationForHost = (endForHost - startForHost) / 1000;
 
+      usageDataObject = new usageData.OfficeAddinUsageData(usageDataOptions);
+
       /* Configure project properties based on user input or answers to prompts */
       this._configureProject(answerForProjectType, answerForScriptType, answerForHost, answerForName, isManifestProject, isExcelFunctionsProject);
-
-      /* Gnerate Insights logging */
-      const noElapsedTime = 0;
-      insight.trackEvent('Name', { Name: this.project.name }, { durationForName });
-      insight.trackEvent('Host', { Host: this.project.host }, { durationForHost });
-      insight.trackEvent('ScriptType', { ScriptType: this.project.scriptType }, { noElapsedTime });
-      insight.trackEvent('IsManifestOnly', { IsManifestOnly: this.project.isManifestOnly.toString() }, { noElapsedTime });
-      insight.trackEvent('ProjectType', { ProjectType: this.project.projectType }, { durationForProjectType });
+      const projectInfo = {
+        Host: [this.project.host, durationForHost],
+        ScriptType: [this.project.scriptType],
+        IsManifestOnly: [this.project.isManifestOnly.toString()],
+        ProjectType: [this.project.projectType, durationForProjectType],
+        isForTesting: [isForTesting]
+      };
+      // Send usage data for project created
+      usageDataObject.reportEvent(defaults.promptSelectionstEventName, projectInfo);
     } catch (err) {
-      insight.trackException(new Error('Prompting Error: ' + err));
+      usageDataObject.reportError(`${defaults.promptSelectionsErrorEventName}${isForTesting ? "-test" : ""}`, new Error('Prompting Error: ' + err));
     }
   },
-
-  writing: function () {
+ 
+  writing: function () { 
     const done = this.async();
     this._copyProjectFiles()
-    .then(() => {
-      done();
-    })
-    .catch((err) => {
-      insight.trackException(new Error('Installation Error: ' + err));
-      process.exitCode = 1;
-    });
+      .then(() => {
+        done();
+      })
+      .catch((err) => {
+        usageDataObject.reportError(`${defaults.copyFilesErrorEventName}${isForTesting ? "-test" : ""}`, new Error('Installation Error: ' + err));
```

Comment:
In case it wasn't clear, I meant to not add "-test" here by the caller, but rather modify the implementation of reportError so that when options.isForTesting is true, it would append "-test". The advantage is that it will be consistently done. It does mean having to make the change in the package, but I think it would be worth doing that.

I presume the desire to use a different event name is because it makes it easier to query for the non "test" events, rather than query all the events where the property value "isForTesting" is false. 

I do think it would be better to have the "isForTesting" option as part of the options passed in, and have the implementation add this to the event data properties automatically. Then you should see the "-test" events have property "isForTesting = true" and the other events should always have isForTesting  = false.

Reply:
I like this suggestion and can make a change in office-addin-usage-data, but this will require that this change pushed and published almost immediately since I will have to bump the version of office-addin-usage-data in generator-office to pick up this change.  Might be worth holding off incorporating this change into this generator-office PR

Reply:
My view is that it would be better to make the update in office-addin-scripts and publish the updated package now, rather than try to make this change later.

Reply:
OK, then this has to get done today - particularly with you being out of town for the next week.  We can interminably have the train idling in the station and things will begin to snowball and next thing we know it's two weeks later and we still haven't wrapped this up.